### ISSUE TITLE
Error message missing name of the failed taskrun

### ISSUE THREAD

**Issue by samueltauil** [tektoncd/pipeline#1142]
# Expected Behavior
Get a error message that provides the name of the task to better troubleshooting purpose.

# Actual Behavior
status:
  completionTime: '2019-07-31T19:54:11Z'
  conditions:
    - lastTransitionTime: '2019-07-31T19:54:11Z'
      message: >-
        invalid output resources: TaskRun's declared resources didn't match
        usage in Task: Didn't provide required values: [source]
      reason: PipelineValidationFailed
      status: 'False'
      type: Succeeded
  startTime: '2019-07-31T19:54:11Z'

# Steps to Reproduce the Problem
Good point, If I knew what Task is failing I could give steps to reproduce.

# Additional Info

**Comment by sbwsg** [comment]
Could you provide the contents of the pipeline/pipelinerun and task/taskrun YAMLs that trigger this?

Looks like one of the tasks is expecting a resource named source and it isn't being wired in.

Taking a wild guess, is the error manifesting from the [tutorial pipeline](https://github.com/tektoncd/pipeline/blob/master/docs/tutorial.md#pipeline)?

**Comment by samueltauil** [comment]
Source for the pipeline and tasks are here: https://github.com/samueltauil/openshift-cd-demo/tree/ocp-4.1-tekton

I don't remember from top of my head a resource called source there but will double check.

I'm starting the pipeline with `tkn pipeline start deploy-pipeline --resource tasks-git=tasks-git --resource tasks-image=tasks-image-internal -s pipeline` 

tkn version: 0.2.0

**Comment by samueltauil** [comment]
I found some references to the `source` reference in my yaml files but still relevant the issue as the errors doesn't show what Task is throwing the error.

**Comment by samueltauil** [comment]
Just add more to the thread, I more concerned about messages like this that doesn't bring the actual Task name that generated that error: 
 `message: >-
        invalid output resources: TaskRun's declared resources didn't match
        usage in Task: Provided extra values: [tasks-git]message: >-`

**Comment by sbwsg** [comment]
Yeah, agreed, it would be helpful to include the name of the Task and TaskRun that caused the mismatch in the error message.

**Comment by EliZucker** [comment]
Looked into this briefly. This isn't an easy fix because those error messages are produced via `ValidateResolvedTaskResources()`. That validation method is reused for TaskRuns and PipelineRuns, both of which drop the particular `Task` name, if it exists (remember they can be directly embedded).

**Comment by EliZucker** [comment]
Additional points: 

- You can find the `TaskRun`/`PipelineRun` name right? I'm assuming that error is attached to one of them. It's still not as easy as it could be, but with that name and the missing parameter's name it's doable to find the source of the error
- We allow embedded `taskSpec` definitions, so in some situations it will be impossible to give the task's name that has a resource validation issue. With the change suggested here we will have 2 different errors, determined by whether the user has embedded tasks or not. I'd say this is a feature at this point rather than a bug

**Comment by samueltauil** [comment]
@EliZucker Yes I do can see a list of TaskRun and PipelineRun but when you have many tasks it is kind of hard to understand where that error came from and looking to the future I'm assuming pipelines are gonna grow either in complexity and size so having this would be a key feature for troubleshooting purposes.

**Comment by dibyom** [comment]
ValidateResolvedTaskResources() has a parameter of type `ResolvedTaskResouces` which has a [`TaskName`](https://github.com/tektoncd/pipeline/blob/master/pkg/reconciler/v1alpha1/taskrun/resources/taskresourceresolution.go#L27). Can we not add that to the error message [here](https://github.com/tektoncd/pipeline/blob/master/pkg/reconciler/v1alpha1/taskrun/validate_resources.go#L121)?

(We'd have to figure out something for embedded tasks but I don't think they are supported in pipelines/pipelineruns at the moment just in taskruns)

**Comment by jerop** [comment]
Duplicate issue: https://github.com/tektoncd/pipeline/issues/2894

Fixed in https://github.com/tektoncd/pipeline/pull/2899

/close
